.github/workflows/sync-3rdparty-images.yaml

```diff
@@ -0,0 +1,104 @@
+name: 'Sync images and artifacts to ghcr'
+on:
+  schedule:
+    - cron: '30 1 * * *'
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  sync-golang:
+    name: 'golang'
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        version:
+          - "1.20"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Tag and push to ghcr
+        run: |
```

Comment:
does 'run' run with 'set -e' by default?

Reply:
Added it